query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -211,7 +211,7 @@ private Expression handleValueConversionForLiteralExpression(Expression lhs, Exp
       return rhs;
     }
 
-    String lhsColumnName = getLogicalColumnName(lhs);
+    String lhsColumnName = getLogicalColumnName(lhs).orElseThrow(IllegalArgumentException::new);
```

Comment:
nit: `orElseThrow()` is probably sufficient throughout, no need to convert them all to `IllegalArgumentException` - without a message it doesn't really provide any extra information.